src/xclim/indices/_hydrology.py

```diff
@@ -685,3 +700,434 @@ def antecedent_precipitation_index(pr: xarray.DataArray, window: int = 7, p_exp:
     out = pr.rolling(time=window).construct("window_dim").dot(weights)
     out.attrs["units"] = "mm"
     return out
+
+
+@declare_units(q="[discharge]", a="[area]", pr="[precipitation]")
+def runoff_ratio(
+    q: xarray.DataArray,
+    a: xarray.DataArray,
+    pr: xarray.DataArray,
+    freq: str = "YS",
+) -> xarray.DataArray:
+    """
+    Runoff ratio.
+
+    Runoff ratio: Ratio of runoff volume measured at the stream to the total
+    precipitation volume over the watershed.
+    Temporal analysis: Yearly values computed from seasonal daily data and yearly data, depending on chosen frequency.
+    (e.g., 'YS' for yearly starting Jan, or 'QS-DEC' for seasons,
+    '30YS' to compute the value over slices of 30 years from the start of the time series).
+
+    Parameters
+    ----------
+    q : xarray.DataArray
+        Streamflow in discharge units. Will be converted to [m³/s].
+    a : xarray.DataArray
+        Watershed area in area units. Will be converted to [km²].
+    pr : xarray.DataArray
+        Mean daily precipitation in precipitation units. Will be converted to [mm/hr].
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray
+        Out: xarray.DataArray
+            Runoff ratio (dimensionless).
+
+    Notes
+    -----
+    - Runoff ratio values are comparable to runoff coefficients.
+    - Values near 0 mean most precipitation infiltrates watershed soil
+      or is lost to evapotranspiration.
+    - Values near 1 mean most precipitation leaves the watershed as runoff.
+      Possible causes are impervious surfaces from urban sprawl, thin soils,
+      steep slopes, etc.
+    - Annual runoff ratios are typically ≤ 1.
+    - Annual runoff ratios are typically higher than summer runoff ratios due to
+      higher levels of evapotranspiration in summer months.
+    - For snow-driven watersheds, spring runoff ratios are typically higher than
+      annual runoff ratios, as snowmelt generates concentrated runoff events.
+
+    References
+    ----------
+    :cite:cts:'knoben_2024'
+    """
+    q = convert_units_to(q, "m3/s")
+    a = convert_units_to(a, "km2")
+    pr = convert_units_to(pr, "mm/hr")
+
+    runoff = q * 3.6 / a  # unit conversion for runoff in mm/hr : 3.6[s/hr *km2/m2]
+    runoff_freq = runoff.resample(time=freq).sum()
+    pr_freq = pr.resample(time=freq).sum()
+
+    out = runoff_freq / pr_freq
+    out.attrs["units"] = ""
+    return out
+
+
+@declare_units(swe="[length]", thresh="[length]")
+def days_with_snowpack(
+    swe: xarray.DataArray,
```

Comment:
Ah I see the nuance now. I would argue that  `swe` is exactly the same as `snw` and the difference in variable doesn't justify a new indicator. Also, `snw` is the CMIP name, which is why we chose it in xclim.